master/github.py

```diff
@@ -0,0 +1,255 @@
+        ('^Configure[-|\s]zfs:(.*)$', 'override-configzfs'),
+    ]
+
+    def parse_comments(self, comments, default_category):
+        category = default_category
+
+        # Extract any overrides for builders for this commit
+        # Requires-builders: style build arch distro test perf none
+        category_pattern = '^Requires-builders:\s*([ ,a-zA-Z0-9]+)'
+        m = re.search(category_pattern, comments, re.I | re.M)
+        if m is not None:
+            category = m.group(1).lower();
+
+            # If Requires-builders contains 'none', then skip this commit
+            none_pattern = '.*none.*'
+            m = re.search(none_pattern, category, re.I | re.M)
+            if m is not None:
+                category = ""
+
+            # If Requires-builders contains 'all', then run all builders.
+            all_pattern = '.*all.*'
+            m = re.search(all_pattern, category, re.I | re.M)
+            if m is not None:
+                category = "style,build,test,perf,coverage,unstable"
+
+        return category
+
+    def handle_push(self, payload):
+        changes = []
+        refname = payload['ref']
+
+        log.msg("Processing GitHub Push `%s'" % refname)
+
+        # We only care about regular heads, i.e. branches
+        match = re.match(r"^refs\/heads\/(.+)$", refname)
+        if not match:
+            log.msg("Ignoring refname `%s': Not a branch" % refname)
+            return changes, 'git'
+
+        branch = match.group(1)
+        if payload.get('deleted'):
+            log.msg("Branch `%s' deleted, ignoring" % branch)
+            return changes, 'git'
+
+        for commit in payload['commits']:
+            if not commit.get('distinct', True):
+                log.msg('Commit `%s` is a non-distinct commit, ignoring...' %
+                        (commit['id'],))
+                continue
+
+            created_at = dateparse(commit['timestamp'])
+            comments = commit['message']
+
+            # Assemble the list of modified files.
+            files = []
+            for kind in ('added', 'modified', 'removed'):
+                files.extend(commit.get(kind, []))
+
+            # Extract if the commit message has property overrides
+            props = { }
+            for prop in CustomGitHubEventHandler.valid_props:
+                step_pattern = prop[0]
+                m = re.search(step_pattern, comments, re.I | re.M)
+                if m is not None:
+                    prop_name = prop[1]
+                    props[prop_name] = json.dumps(m.group(1).lower())
+
+            # Extract if the commit message has property overrides
+            category = self.parse_comments(comments,
+                "style,build,test,perf,coverage,unstable")
+
+            # Releases prior to 0.8.0 required an external spl build.
+            match = re.match(r".*-0.[0-7]-release", branch)
+            if not match:
+                props['buildspl'] = json.dumps("no")
```

Comment:
I think `json.dumps` is causing quotes around the various properties. Was that intentional?

Reply:
Yes, it's intentional but undesirable.  The change properties are expected to be stored as json in the database.  When the change is scheduled the property value is assigned with `json.loads` which seems to preserve the quotes.  From what I can tell this seems to be a known python 2.7 issue which was fixed in newer versions of python.

I wasn't able to find a way to prevent this, so I figured we could live with it for now.  But if you know how that'd be great!

Reply:
Bah, that's unfortunate. If we wanted to fix it, we'd have to introduce another buildbot patch to fix it around the `json.loads` call is my guess.